src/main/scala/org/neo4j/spark/utils/Neo4jUtils.scala

```diff
@@ -0,0 +1,58 @@
+package org.neo4j.spark.utils
+import java.util.concurrent.Callable
+import java.util.function
+
+import org.neo4j.driver.{Driver, Result, Session, Transaction}
+import io.github.resilience4j.retry.{Retry, RetryConfig}
+import org.neo4j.driver.exceptions.{ServiceUnavailableException, SessionExpiredException, TransientException}
+import org.neo4j.spark.Neo4jConfig
+
+
+object Neo4jUtils {
+
+  def close(driver: Driver, session: Session): Unit = {
+    try {
+      if (session != null && session.isOpen) {
+        session.close()
+      }
+    } finally {
+      if (driver != null) {
+        driver.close()
+      }
+    }
+  }
+
+  private val retryConfig = RetryConfig.custom.retryExceptions(
```

Comment:
wouldn't that be usually be something that's handled by Spark? How do others handle this? Just curious.

Reply:
The spark retry mechanism works with `spark.task.maxFailures` which is:
```
Number of failures of any particular task before giving up on the job.
The total number of failures spread across different tasks will not cause the job to fail;
a particular task has to fail this number of attempts
```

But IMHO the control it's not fine-graded as this one, I would like to have both of them, wdyt?

Reply:
perhaps make it configurable? (later PR)